### PR TITLE
Sav - Set tasks tab default unless user has completed all tasks

### LINKED ISSUE #6: Set permissions for time log.

### PR DESCRIPTION
**Pull request by Lunaliar**
# Description
While working on the task stay default tab bug I found a small bug where the tasks tab would still be shown even if a user had completed all of their tasks. I added a small conditional to check the status of the tasks when the dashboard loads. Just a small adjustment to handle an edge case.  This fix should only apply to volunteer accounts as admin accounts will always default to the tasks tab on the dashboard.
Fixes #6  (Reports Component  priority low)

## Related PRS (if any):
This frontend PR is related to the #686 frontend PR.

## Mainly changes explained:
Added secondary conditional to check the status of tasks in userTasks array.

## How to test:
Check into the current branch
Do `npm install` and `...` to run this PR locally
Log in as an admin user
Go to Projects -> WBS -> make three tasks for a volunteer. 
Set each one of the tasks to ‘not started’, ‘started’, and ‘complete’
Log in to volunteer acct.
Take note of dashboard load behavior. Is the task tab showing as default with the user-assigned tasks?
If you delete tasks or make them all complete it should show the current week time log tab by default on the dashboard.

## Screenshots or videos of changes:
complete status

## Note:
I was able to quickly test this by opening a second terminal for the front end and running it on port 3001 to have two windows. One for the admin and one for the volunteer.

### CODE REVIEW

**Comment by one-community** [comment]
Excellent, thank you

**Review by vitorsaa2k — APPROVED** [review]
Tested and when all my tasks are completed the default tab is the Current Time Log. When the tasks are not completed the default tab is the Tasks tab.

**Review by ToobaJamal — APPROVED** [review]
I have reviewed and tested the changes. When tasks are assigned to a volunteer, the Tasks tab is set to default and when all tasks are completed, the Current Week Timelog tab is the default. 
With tasks assigned
withTasksAssigned

With tasks completed/ deleted

withTasksDeleted

**Comment by one-community** [comment]
Thank you all, moving this to final review.

**Review by abdel-lall — APPROVED** [review]
the changes work as described. nice job!

PR734

**Comment by one-community** [comment]
Thank you all, moving this to final review!

File: src/components/Timelog/Timelog.jsx
Comment (EvianTan) [inline src/components/Timelog/Timelog.jsx:52]: Please remove this debug output.

**Comment by Lunaliar** [comment]
Sorry about that, the debug statement has been removed!

File: src/components/Timelog/Timelog.jsx
Comment (Vishvesh-Sheoran) [inline src/components/Timelog/Timelog.jsx:1]: The following version uses a _for...of_ loop to iterate over each task in the _tasks_ array. It then checks if the current task has a WBS ID and a non-complete status. If it finds a task that meets these criteria, it immediately returns _true_. If it reaches the end of the loop without finding any matching tasks, it returns _false_.

By using an early return, we can avoid unnecessary iterations through the loop once we have found the result. This code is also more concise and easier to read than the original versions:

const doesUserHaveTaskWithWBS = (tasks) => {
  for (let task of tasks) {
    if (task.wbsId && task.status !== 'Complete') {
      return true;
    }
  }
  return false;
};

**Review by EvianTan — APPROVED** [review]

**Comment by one-community** [comment]
Will merge once Vish's code suggestions are addressed/integrated. If you feel they don't apply, please explain why. Be sure to check the basic function again too, if there's any chance of those code changes impacting/affecting it.

**Comment by AnishPandita** [comment]
Hi @Lunaliar I tested the PR and its working as intended. The tab switch behaviour is correct according to the current status of task completion.
Screenshot 2023-04-27 at 6 52 13 PM
Screenshot 2023-04-27 at 6 54 25 PM

**Comment by one-community** [comment]
Tested this and accounts other than Volunteer are now reverting to Timelog tab too. Not all, but some. Please raise a new PR that fixes this so ONLY Volunteer accounts display this function.

**Comment by one-community** [comment]
Actually, please leave this to someone else because the badges task you are working on is so huge and way more important. If you'd like to have help with that, let me know too.

**Pull request merged**
